i18n/ru/docusaurus-plugin-content-docs/current/guides/examples/api-requests.mdx

```diff
@@ -0,0 +1,140 @@
+
+export const client = axios.create({
+  baseURL: 'https://your-api-domain.com/api/',
+  timeout: 5000,
+  headers: { 'X-Custom-Header': 'my-custom-value' }
+});
+```
+</TabItem>
+
+<TabItem value="fetch" label="Fetch">
+```ts title="shared/api/client.ts"
+export const client = {
+  async post(endpoint: string, body: any, options?: RequestInit) {
+    const response = await fetch(`https://your-api-domain.com/api${endpoint}`, {
+      method: 'POST',
+      body: JSON.stringify(body),
+      ...options,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Custom-Header': 'my-custom-value',
+        ...options?.headers,
+      },
+    });
+    return response.json();
+  }
+  // ... другие методы put, delete, и т.д.
+};
+```
+</TabItem>
+
+</Tabs>
+
+Организуйте свои отдельные функции API-запросов в shared/api/endpoints, группируя их по API эндпоинтам.
+
+:::note
+
+Для простоты, в примерах ниже мы опускаем взаимодействие с формами и валидацию. Для получения подробной информации о том как работать с такими библиотеками как Zod или Valibot, обратитесь к статье [Проверка типов и схемы](/docs/guides/examples/types#type-validation-schemas-and-zod).
+
+:::
+
+```ts title="shared/api/endpoints/login.ts"
+import { client } from '../client';
+
+export interface LoginCredentials {
+  email: string;
+  password: string;
+}
+
+export function login(credentials: LoginCredentials) {
+  return client.post('/login', credentials);
+}
+```
+
+Используйте файл `index.ts` в `shared/api` для экспорта ваших функций запросов.
+
+```ts title="shared/api/index.ts"
+export { client } from './client'; // Если нужно экспортировать клиент
+export { login } from './endpoints/login';
+export type { LoginCredentials } from './endpoints/login';
+```
+
+## API-запросы, специфичные для слайса {#client-generators}
+
+Если API-запрос используется только определенным слайсом (например, одной страницей или функцией) и не будет использоваться повторно, поместите его в сегмент api этого слайса. Это позволит аккуратно хранить логику, специфичную для слайса.
+
+- 📂 pages
+    - 📂 login
+        - 📄 index.ts
+        - 📂 api
+            - 📄 login.ts
+        - 📂 ui
+            - 📄 LoginPage.tsx
+
+```ts title="pages/login/api/login.ts"
+import { client } from 'shared/api';
+
+interface LoginCredentials {
+  email: string;
+  password: string;
+}
+
+export function login(credentials: LoginCredentials) {
+  return client.post('/login', credentials);
+}
+```
+
+Вам не нужно экспортировать функцию `login()` через публичный API страницы, потому что маловероятно, что какое-либо другое место в приложении будет нуждаться в этом запросе.
+
+:::note
+
+Избегайте преждевременного размещения вызовов API и типов ответов в слое `entities`. Ответы бэкенда могут отличаться от того, что нужно вашим сущностям фронтенда. Логика API в `shared/api` или сегменте `api` слайса позволяет вам преобразовывать данные при необходимости, сохраняя фокус сущностей на проблемах фронтенда.
```

Comment:
suggestion: clarity

```suggestion
Избегайте преждевременного размещения API-запросов и типов ответов бэкенда в слое `entities`. Ответы бэкенда могут отличаться от того, что нужно вашим сущностям фронтенда. Логика API в `shared/api` или сегменте `api` слайса позволяет вам преобразовывать данные при необходимости, сохраняя фокус сущностей на проблемах фронтенда.
```

Reply:
There are a lot of reasons why this type of code should not go to the `entities`, and it might be a huge topic on its own. But in order to make this guide compact and easy to follow, I just decided to go with this one

I'll try to make a separate guide on entities where we can go into the details, but for now I think it is better to keep `en` and `ru` versions in sync